src/date-range-picker/index.tsx

```diff
@@ -288,14 +298,21 @@ const DateRangePicker = React.forwardRef(
 
     const mergedRef = useMergeRefs(rootRef, __internalRootRef);
 
+    useEffect(() => {
```

Comment:
I would like to iterate on if this useEffect hook is needed. What is does is parse the initial formatting that might not reflect the dateOnly or a month granularity.

Reply:
Ok.  I looked into this because the 'RAW' value on the pages was equal to the initial values, whereas the values in the input were formatted. I need to look into if this is expected? Say this input is on a form, and that raw value is what gets passed as form data on a submission as it was never formatted to match a month granularity or dateOnly expectation, as it would be if component was interacted with.